backend/src/database/users.go

```diff
@@ -35,19 +41,28 @@ func (db *DB) GetCurrentUsers(page, itemsPerPage int, facilityId uint, order str
 	return count, users, nil
 }
 
-func (db *DB) SearchCurrentUsers(page, itemsPerPage int, facilityId uint, order, search string) (int64, []models.User, error) {
+func (db *DB) SearchCurrentUsers(page, itemsPerPage int, facilityId uint, order, search string, role string) (int64, []models.User, error) {
 	var users []models.User
 	var count int64
 	offset := (page - 1) * itemsPerPage
 	search = strings.TrimSpace(search)
-	likeSearch := "%" + search + "%"
-	if err := db.Model(models.User{}).
-		Where("facility_id = ?", fmt.Sprintf("%d", facilityId)).
-		Where("name_first ILIKE ? OR username ILIKE ? OR name_last ILIKE ?", likeSearch, likeSearch, likeSearch).
+	likeSearch := "%" + strings.ToLower(search) + "%"
+	tx := db.Model(&models.User{})
+	switch role {
+	case "admin":
+		tx = tx.Where("role = 'admin'")
+	case "student":
+		tx = tx.Where("role = 'student'")
+	}
+	if err := tx.Count(&count).Error; err != nil {
+		return 0, nil, newGetRecordsDBError(err, "users")
+	}
+	if err := tx.
+		Find(&users, "facility_id = ? AND (name_first LIKE ? OR username LIKE ? OR name_last LIKE ?)", facilityId, likeSearch, likeSearch, likeSearch).
 		Order(order).
 		Offset(offset).
 		Limit(itemsPerPage).
-		Find(&users).Count(&count).Error; err != nil {
+		Error; err != nil {
```

Comment:
I think this should still include `.Count(&count)`, so the complete line should be `.Count(&count).Error; err != nil {`